### PR TITLE
Ieee 211 general popup modal new version

### PR DESCRIPTION
**Pull request by ericji1326**
## Overview

- Resolves IEEE211
- Refer to old version of ticket

Note: Admins Team page was modified to be able to test the modal, before merging I'll revert this change

### CODE REVIEW

File: hackathon_site/dashboard/frontend/src/components/general/PopupModal/PopupModal.test.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
+import { render } from "../../../testing/utils";
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/components/general/PopupModal/PopupModal.test.tsx]: relative import here, change to absolute

File: hackathon_site/dashboard/frontend/src/pages/Teams/Teams.tsx [hunk 2]
```diff
@@ -1,5 +1,43 @@
-import React from "react";
-// import styles from "./Teams.module.scss";
+// import React from "react";
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/pages/Teams/Teams.tsx]: please remove these comments and remove your testing popup modal. If you'd like to implement your modal you could do it for cancel order on the participant side, check line 216 (PendingTables) in ItemTable.tsx to see the cancel order button.

File: hackathon_site/dashboard/frontend/src/components/general/PopupModal/PopupModal.module.scss [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+.modal {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+}
+
+.modalGrid {
+    background-color: white;
+    padding: 1em;
+    @include responsive(sm-down) {
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/components/general/PopupModal/PopupModal.module.scss:15]: for mobile screens, set screen size to a variable width, 85% looked good to me while I was testing. Also move your @include statement to the bottom of the class, it just looks better that way.

File: hackathon_site/dashboard/frontend/src/components/general/PopupModal/PopupModal.module.scss [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        width: 200px;
+    }
+    width: 350px;
+}
+
+.title {
+    text-align: center;
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/components/general/PopupModal/PopupModal.module.scss:21]: add font-size:25px here, the extremely large title just didn't look right to me 😅

File: hackathon_site/dashboard/frontend/src/components/general/PopupModal/PopupModal.test.tsx [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+test("cancel and submit functions are NOT being passed into popup modal without crashing", () => {
+    const { getByText } = render(
+        <PopupModal
+            description={"Example Description"}
+            isVisible={true}
+            cancelText={"cancel"}
+            submitText={"submit"}
+            title={"Test Modal"}
+        ></PopupModal>
+    );
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/components/general/PopupModal/PopupModal.test.tsx]: you should add an assertion here testing that the cancel and submit button don't appear on the modal

File: hackathon_site/dashboard/frontend/src/components/general/PopupModal/PopupModal.test.tsx [hunk 6]
```diff
@@ -0,0 +1,85 @@
+import { render } from "../../../testing/utils";
+import PopupModal from "./PopupModal";
+import React from "react";
+import { fireEvent } from "@testing-library/react";
+
+test("renders general popup modal without crashing", () => {
+    const { getByText } = render(
+        <PopupModal
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/components/general/PopupModal/PopupModal.test.tsx:8]: use a self-closing tag for PopupModal like this: `<PopupModal />`

**Review by YuYing-Liang — COMMENTED** [review]
Some last comments, so far it's looking good

File: hackathon_site/dashboard/frontend/src/components/general/PopupModal/PopupModal.test.tsx (hunk 1 above)
Comment (ericji1326) [reply hackathon_site/dashboard/frontend/src/components/general/PopupModal/PopupModal.test.tsx]: done

File: hackathon_site/dashboard/frontend/src/components/general/PopupModal/PopupModal.module.scss (hunk 3 above)
Comment (ericji1326) [reply hackathon_site/dashboard/frontend/src/components/general/PopupModal/PopupModal.module.scss:15]: done

File: hackathon_site/dashboard/frontend/src/components/general/PopupModal/PopupModal.module.scss (hunk 4 above)
Comment (ericji1326) [reply hackathon_site/dashboard/frontend/src/components/general/PopupModal/PopupModal.module.scss:21]: alright

File: hackathon_site/dashboard/frontend/src/components/general/PopupModal/PopupModal.test.tsx (hunk 5 above)
Comment (ericji1326) [reply hackathon_site/dashboard/frontend/src/components/general/PopupModal/PopupModal.test.tsx]: done

**Review by YuYing-Liang — DISMISSED** [review]
I know you'll be working on this more so could you add the following:
- make the cancel handler required (idk why I said it was optional before bc you can't close the modal if you don't have the cancel function there lol)
- put a small border radius on the popup, 5 or 10px should be enough (only bc most boxes on the site have a small border radius so it makes the popup look really sharp)

File: hackathon_site/dashboard/frontend/src/components/teamDetail/TeamCheckedOutOrderTable/TeamCheckedOutOrderTable.test.tsx [hunk 7]
```diff
@@ -0,0 +1,102 @@
+import React from "react";
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/components/teamDetail/TeamCheckedOutOrderTable/TeamCheckedOutOrderTable.test.tsx:1]: something's off here, could you merge with develop?

Comment (YuYing-Liang) [reply hackathon_site/dashboard/frontend/src/components/teamDetail/TeamCheckedOutOrderTable/TeamCheckedOutOrderTable.test.tsx:1]: These changes shouldn't be showing up

Comment (ericji1326) [reply hackathon_site/dashboard/frontend/src/components/teamDetail/TeamCheckedOutOrderTable/TeamCheckedOutOrderTable.test.tsx:1]: I did a git pull --rebase with develop and this happened...

**Review by YuYing-Liang — APPROVED** [review]

**Pull request merged**